test/components/rankers/test_sentence_transformers_diversity.py

```diff
@@ -0,0 +1,394 @@
+
+        with pytest.raises(ComponentError):
+            ranker.run(query="test query", documents=documents)
+
+    @pytest.mark.integration
+    @pytest.mark.parametrize("similarity", ["dot_product", "cosine"])
+    def test_run_empty_query(self, similarity):
+        """
+        Test that ranker can be run with an empty query.
+        """
+        ranker = SentenceTransformersDiversityRanker(
+            model="sentence-transformers/all-MiniLM-L6-v2", top_k=3, similarity=similarity
+        )
+        ranker.warm_up()
+        documents = [Document(content="doc1"), Document(content="doc2")]
+
+        result = ranker.run(query="", documents=documents)
+        ranked_docs = result["documents"]
+
+        assert isinstance(ranked_docs, list)
+        assert len(ranked_docs) == 2
+        assert all(isinstance(doc, Document) for doc in ranked_docs)
+
+    @pytest.mark.integration
+    @pytest.mark.parametrize("similarity", ["dot_product", "cosine"])
+    def test_run_top_k(self, similarity):
+        """
+        Test that run method returns the correct number of documents for different top_k values passed at
+        initialization and runtime.
+        """
+        ranker = SentenceTransformersDiversityRanker(
+            model="sentence-transformers/all-MiniLM-L6-v2", similarity=similarity, top_k=3
+        )
+        ranker.warm_up()
+        query = "test query"
+        documents = [
+            Document(content="doc1"),
+            Document(content="doc2"),
+            Document(content="doc3"),
+            Document(content="doc4"),
+        ]
+
+        result = ranker.run(query=query, documents=documents)
+        ranked_docs = result["documents"]
+
+        assert isinstance(ranked_docs, list)
+        assert len(ranked_docs) == 3
+        assert all(isinstance(doc, Document) for doc in ranked_docs)
+
+        # Passing a different top_k at runtime
+        result = ranker.run(query=query, documents=documents, top_k=2)
+        ranked_docs = result["documents"]
+
+        assert isinstance(ranked_docs, list)
+        assert len(ranked_docs) == 2
+        assert all(isinstance(doc, Document) for doc in ranked_docs)
+
+    @pytest.mark.integration
+    @pytest.mark.parametrize("similarity", ["dot_product", "cosine"])
+    def test_run_negative_top_k(self, similarity):
+        """
+        Tests that run method raises an error for negative top-k.
+        """
+        ranker = SentenceTransformersDiversityRanker(
+            model="sentence-transformers/all-MiniLM-L6-v2", similarity=similarity, top_k=10
+        )
+        ranker.warm_up()
+        query = "test"
+        documents = [Document(content="doc1"), Document(content="doc2"), Document(content="doc3")]
+
+        # Setting top_k at runtime
+        with pytest.raises(ValueError):
+            ranker.run(query=query, documents=documents, top_k=-5)
+
+        # Setting top_k at init
+        with pytest.raises(ValueError):
```

Comment:
Can we take this test out of here so we can make it a standalone unit test. We tend to prefer make each test as small as possible and make things unit tests where possible. We only count code coverage with unit tests currently and not with integration tests.

Reply:
I understand. I will refactor this test and move this to a separate unit test.